From: Aidas Jonikas <aidas957@gmail.com>
Subject: [PATCH] dxdiagn: Update DirectX version to the one from Windows 11.
Message-Id: <20211202113537.3559446-1-aidas957@gmail.com>
Date: Thu,  2 Dec 2021 13:35:37 +0200

This patch removes the DirectX 9 dialog in Roblox that prevents 
the game from starting and/or installing.

Technically only DirectX 11 is required for fixing the issue 
but I want to future-proof this patch.
---
 dlls/dxdiagn/provider.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/dlls/dxdiagn/provider.c b/dlls/dxdiagn/provider.c
index 11111111111..11111111111 100644
--- a/dlls/dxdiagn/provider.c
+++ b/dlls/dxdiagn/provider.c
@@ -614,7 +614,7 @@ static HRESULT build_systeminfo_tree(IDxDiagContainerImpl_Container *node)
     WCHAR buffer[MAX_PATH], computer_name[MAX_COMPUTERNAME_LENGTH + 1], print_buf[200], localized_pagefile_fmt[200];
     DWORD_PTR args[2];
 
-    hr = add_ui4_property(node, L"dwDirectXVersionMajor", 9);
+    hr = add_ui4_property(node, L"dwDirectXVersionMajor", 12);
     if (FAILED(hr))
         return hr;
 
@@ -626,11 +626,11 @@ static HRESULT build_systeminfo_tree(IDxDiagContainerImpl_Container *node)
     if (FAILED(hr))
         return hr;
 
-    hr = add_bstr_property(node, L"szDirectXVersionEnglish", L"4.09.0000.0904");
+    hr = add_bstr_property(node, L"szDirectXVersionEnglish", L"10.00.22000.1000");
     if (FAILED(hr))
         return hr;
 
-    hr = add_bstr_property(node, L"szDirectXVersionLongEnglish", L"= \"DirectX 9.0c (4.09.0000.0904)");
+    hr = add_bstr_property(node, L"szDirectXVersionLongEnglish", L"= \"DirectX 12 (10.00.22000.1000)");
     if (FAILED(hr))
         return hr;
 

-- 
2.35.3

